app/models/redirection.rb

```diff
@@ -0,0 +1,16 @@
+class Redirection < ActiveRecord::Base
+  belongs_to :next, class_name: "Redirection"
+  has_one :previous, class_name: "Redirection", foreign_key: "next_id"
+
+  validates :next_id, uniqueness: true, allow_blank: true
+  validates :slug, presence: true, uniqueness: true
+  validates :url, presence: true
+
+  def next_url
+    self.next.url
```

Comment:
It's not in previous (just removed it), but it is in next because `next` is a ruby keyword